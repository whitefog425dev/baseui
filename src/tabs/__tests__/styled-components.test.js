/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledTabBar,
  StyledTabContent,
  StyledTab,
  ORIENTATION,
} from '../index.js';

const styledComponents = [
  [StyledRoot, 'StyledRoot'],
  [StyledTabBar, 'StyledTabBar'],
  [StyledTabContent, 'StyledTabContent'],
  [StyledTab, 'StyledTab'],
];

describe.each(Object.values(ORIENTATION))(
  'Tabs styled components in %s orientation',
  orientation => {
    test.each(styledComponents)('Basic render', (Component, name) => {
      const component = shallow(<Component $orientation={orientation} />);
      expect(component.instance().getStyles()).toMatchSnapshot(
        `${name} has correct styles in ${orientation} orientation`,
      );
    });

    test.each(styledComponents)(
      'Basic render when disabled',
      (Component, name) => {
        const component = shallow(
          <Component $orientation={orientation} $disabled={true} />,
        );
        expect(component.instance().getStyles()).toMatchSnapshot(
          `${name} has correct styles when disabled is set to true in ${orientation} orientation`,
        );
      },
    );

    test.each([
      [StyledTab, 'StyledTab'],
      [StyledTabContent, 'StyledTabContent'],
    ])('Basic render when active is set to true', (Component, name) => {
      const component = shallow(
        <Component $orientation={orientation} $active={true} />,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        `${name} has correct styles when active is set to true in ${orientation} orientation`,
      );
    });
  },
);
