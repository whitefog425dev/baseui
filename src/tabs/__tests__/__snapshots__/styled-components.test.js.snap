// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs styled components in horizontal orientation Basic render when active is set to true: StyledTab has correct styles when active is set to true in horizontal orientation 1`] = `
Object {
  ":focus": Object {
    "color": "$theme.colors.primary400",
  },
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "borderBottom": "2px solid $theme.colors.primary400",
  "boxSizing": "border-box",
  "color": "$theme.colors.black",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render when active is set to true: StyledTabContent has correct styles when active is set to true in horizontal orientation 1`] = `
Object {
  "display": "block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render when disabled: StyledRoot has correct styles when disabled is set to true in horizontal orientation 1`] = `
Object {
  "display": "flex",
  "flexDirection": "column",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render when disabled: StyledTab has correct styles when disabled is set to true in horizontal orientation 1`] = `
Object {
  "borderBottom": "none",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono800",
  "cursor": "not-allowed",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render when disabled: StyledTabBar has correct styles when disabled is set to true in horizontal orientation 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "flexDirection": "row",
  "paddingLeft": "$theme.sizing.scale400",
  "paddingRight": "$theme.sizing.scale400",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render when disabled: StyledTabContent has correct styles when disabled is set to true in horizontal orientation 1`] = `
Object {
  "display": "none",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render: StyledRoot has correct styles in horizontal orientation 1`] = `
Object {
  "display": "flex",
  "flexDirection": "column",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render: StyledTab has correct styles in horizontal orientation 1`] = `
Object {
  ":focus": Object {
    "color": "$theme.colors.primary400",
  },
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "borderBottom": "none",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono800",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render: StyledTabBar has correct styles in horizontal orientation 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "flexDirection": "row",
  "paddingLeft": "$theme.sizing.scale400",
  "paddingRight": "$theme.sizing.scale400",
}
`;

exports[`Tabs styled components in horizontal orientation Basic render: StyledTabContent has correct styles in horizontal orientation 1`] = `
Object {
  "display": "none",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when active is set to true: StyledTab has correct styles when active is set to true in vertical orientation 1`] = `
Object {
  ":focus": Object {
    "color": "$theme.colors.primary400",
  },
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "borderBottom": "none",
  "boxSizing": "border-box",
  "color": "$theme.colors.black",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when active is set to true: StyledTabContent has correct styles when active is set to true in vertical orientation 1`] = `
Object {
  "display": "block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when disabled: StyledRoot has correct styles when disabled is set to true in vertical orientation 1`] = `
Object {
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when disabled: StyledTab has correct styles when disabled is set to true in vertical orientation 1`] = `
Object {
  "borderBottom": "none",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono800",
  "cursor": "not-allowed",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when disabled: StyledTabBar has correct styles when disabled is set to true in vertical orientation 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "flexDirection": "column",
  "paddingLeft": "$theme.sizing.scale400",
  "paddingRight": "$theme.sizing.scale400",
}
`;

exports[`Tabs styled components in vertical orientation Basic render when disabled: StyledTabContent has correct styles when disabled is set to true in vertical orientation 1`] = `
Object {
  "display": "none",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Tabs styled components in vertical orientation Basic render: StyledRoot has correct styles in vertical orientation 1`] = `
Object {
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`Tabs styled components in vertical orientation Basic render: StyledTab has correct styles in vertical orientation 1`] = `
Object {
  ":focus": Object {
    "color": "$theme.colors.primary400",
  },
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "borderBottom": "none",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono800",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`Tabs styled components in vertical orientation Basic render: StyledTabBar has correct styles in vertical orientation 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "display": "flex",
  "flexDirection": "column",
  "paddingLeft": "$theme.sizing.scale400",
  "paddingRight": "$theme.sizing.scale400",
}
`;

exports[`Tabs styled components in vertical orientation Basic render: StyledTabContent has correct styles in vertical orientation 1`] = `
Object {
  "display": "none",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale500",
}
`;
